src/main/java/com/synopsys/integration/alert/channel/jira/util/JiraIssuePropertyHelper.java

```diff
@@ -0,0 +1,90 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.channel.jira.util;
+
+import java.util.Optional;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.synopsys.integration.alert.channel.jira.JiraConstants;
+import com.synopsys.integration.alert.channel.jira.model.AlertJiraIssueProperties;
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.jira.common.cloud.model.response.IssueSearchResponseModel;
+import com.synopsys.integration.jira.common.cloud.rest.service.IssuePropertyService;
+import com.synopsys.integration.jira.common.cloud.rest.service.IssueSearchService;
+
+public class JiraIssuePropertyHelper {
+    private final IssueSearchService issueSearchService;
+    private final IssuePropertyService issuePropertyService;
+
+    public JiraIssuePropertyHelper(IssueSearchService issueSearchService, IssuePropertyService issuePropertyService) {
+        this.issueSearchService = issueSearchService;
+        this.issuePropertyService = issuePropertyService;
+    }
+
+    public Optional<IssueSearchResponseModel> findIssues(String category, String bomComponentUri) throws IntegrationException {
+        return findIssues(category, bomComponentUri, null);
+    }
+
+    public Optional<IssueSearchResponseModel> findIssues(String category, String bomComponentUri, String policyName) throws IntegrationException {
+        final StringBuilder jqlBuilder = new StringBuilder();
+        if (StringUtils.isNotBlank(category)) {
+            jqlBuilder.append(createPropertySearchString(JiraConstants.JIRA_ISSUE_PROPERTY_OBJECT_KEY_CATEGORY, category));
+            jqlBuilder.append(StringUtils.SPACE);
+        }
+        if (StringUtils.isNotBlank(bomComponentUri)) {
+            jqlBuilder.append(StringUtils.SPACE);
+            jqlBuilder.append(createPropertySearchString(JiraConstants.JIRA_ISSUE_PROPERTY_OBJECT_KEY_BOM_COMPONENT_URI, bomComponentUri));
+        }
+        if (StringUtils.isNotBlank(policyName)) {
+            jqlBuilder.append(StringUtils.SPACE);
+            jqlBuilder.append(createPropertySearchString(JiraConstants.JIRA_ISSUE_PROPERTY_OBJECT_KEY_POLICY_NAME, policyName));
+        }
+
+        final String jql = jqlBuilder.toString();
+        if (!jql.isBlank()) {
+            final IssueSearchResponseModel issueSearchResponseModel = issueSearchService.queryForIssues(jql);
+            return Optional.of(issueSearchResponseModel);
+        }
+        return Optional.empty();
+    }
+
+    public void addPropertiesToIssue(String issueKey, String category, String bomComponentUri) throws IntegrationException {
```

Comment:
I think We were going to use one of the content keys as the property key now to still be unique and also independent of providers